File: Qytetet Java/src/modeloqytetet/Sorpresa.java

package modeloqytetet;

/**
 *
 * @author miguel
 */
public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    
    public Sorpresa(String texto,int valor, TipoSorpresa tipo){
        this.texto=texto;
        this.tipo=tipo;
        this.valor=valor;
    }
    
    public String getTexto(){
        return texto;
    }
    
    public TipoSorpresa getTipo(){
        return tipo;
    }
    
    public int getValor(){
        return valor;
    }
    
    //Este metodo devuelve un string con el estado del objeto correspondiente
    public String toString(){
        return "Sorpresa{"+"texto="+texto + ",valor="+
                Integer.toString(valor) + ",tipo="+tipo + "}";
    }
           
}

File: Qytetet Ruby/pkg/QytetetRuby-0.0.1/lib/sorpresa.rb
#encoding: utf-8
module ModeloQytetet
  class Sorpresa
    def initialize (texto,valor,tipo)
      @texto=texto
      @valor=valor
      @tipo=tipo
    end

    attr_reader :texto
    attr_reader :valor 
    attr_reader :tipo


    def to_s
      "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end


  end
end